app/code/Magento/AdvancedPricingImportExport/Model/Import/AdvancedPricing.php

```diff
@@ -591,8 +595,10 @@ protected function incrementCounterUpdated($prices, $existingPrice)
         foreach ($prices as $price) {
             if ($existingPrice['all_groups'] == $price['all_groups']
                 && $existingPrice['customer_group_id'] == $price['customer_group_id']
+                && (int) $existingPrice['qty'] == (int) $price['qty']
```

Comment:
In case if you use a typecasting, it's a good idea to compare the values using strict comparison (`===`).

Reply:
In this case we have different strings.
For example 
$existingPrice['qty'] is '9.0' and $price['qty'] is '9.0000'
the strict comparison return "false"
So, I decided to play it safe and use the int type.

Reply:
I mean using both typecasting and strict comparison. So in result we will have:

```
(int) $existingPrice['qty'] === (int) $price['qty']
```